src/Frontend/Page/User.hs

```diff
@@ -252,40 +259,61 @@ instance ToHtml PageUserProfileDelegatedVotes where
                     -- School / Class select buttons: FIXME mechanics!
                     div_ [class_ "filter-toggles"] $ do
-                        button_ [class_ "filter-toggle-btn", value_ ""] "Schulweit"
-                        button_ [class_ "filter-toggle-btn m-active", value_ ""] "Klassenweit"
+                        a_ [class_ "filter-toggle-btn", href_ (U.userGlobalDelegations user)] "Schulweit"
+                        a_ [class_ "filter-toggle-btn m-active", href_ (U.userClassDelegations user)] "Klassenweit"
                     renderDelegations delegations
 
-renderDelegations :: forall m. Monad m => [Delegation] -> HtmlT m ()
-renderDelegations _ = do
+renderDelegations :: forall m. Monad m => DelegationInfo -> HtmlT m ()
+renderDelegations delegations = do
     h2_ $ "Insgesamt " <> total ^. showed . html
-    ul_ [class_ "small-avatar-list"] $ renderLi `mapM_` [undefined, undefined, undefined]  -- FIXME
+    ul_ [class_ "small-avatar-list"] $ renderLi `mapM_` delegations
   where
-    total :: Int
-    total = 20
+    total = sum $ map ((1 +) . length . snd) delegations
 
-    renderLi :: Delegation -> HtmlT m ()  -- FIXME
-    renderLi _ = do
+    renderLi :: (User, [User]) -> HtmlT m ()
+    renderLi (delegatee, secondDelegatees) = do
         li_ [class_ "small-avatar-list-item"] $ do
             div_ [class_ "col-1-12"] $ do
                 div_ [class_ "small-avatar-list-image"] $ do
                     nil -- FIXME Make a real image a child here (avatarImgFromHasMeta)
             div_ [class_ "col-11-12"] $ do
-                h3_ "UserName"
+                h3_ $ a_ [href_ $ U.viewUserProfile delegatee] (delegatee ^. userLogin . unUserLogin  . html)
                 p_ $ do
-                    "5 Stimmen von "
-                    strong_ $ do
-                        a_ [href_ U.Broken] "UserName, "
-                        a_ [href_ U.Broken] "UserName, "
-                        a_ [href_ U.Broken] "UserName"
+                    toHtml $ show (length secondDelegatees) <> " Stimmen von "
+                    strong_ . forM_ secondDelegatees $ \delegatee' ->
+                        a_ [href_ $ U.viewUserProfile delegatee'] (delegatee' ^. userLogin . unUserLogin  . html)
 
-delegatedVotes :: (ActionPersist m, ActionUserHandler m)
+delegatedVotesGlobal :: (ActionPersist m, ActionUserHandler m)
       => AUID User -> m PageUserProfileDelegatedVotes
-delegatedVotes userId = do
-    PageUserProfileDelegatedVotes
-    <$> currentUserCapCtx
-    <*> (makeUserView <$> mquery (findUser userId))
-    <*> pure [] -- FIXME
+delegatedVotesGlobal userId = delegatedVotes userId DScopeGlobal
+
+delegatedVotesClass :: (ActionPersist m, ActionUserHandler m)
+      => AUID User -> m PageUserProfileDelegatedVotes
+delegatedVotesClass userId = do
+    user <- mquery (findUser userId)
+    case user ^? userRole . _Student of
+        -- TODO: Translation
+        Nothing -> throwError500 "User is not a student"
```

Comment:
I'm on @fisx side here, internal errors do not have to treated as part of the UI, hence no need to translate them.